### PR TITLE
extract events from the discharged_concept_id field of the visit table with the correct time

### PR DESCRIPTION
**Pull request by ChaoPang**
We want to extract discharged_concept_id as a separate event from the visit table. This will be important for identifying hospital discharge events when using ACES.

### CODE REVIEW

File: src/meds_etl/omop.py [hunk 1]
```diff
@@ -188,6 +194,8 @@ def write_event_data(
                 for option in options
                 if table_name + option in schema.names()
             ]
+            # We prefer user defined time field options over the default time options if available
```
Comment (EthanSteinberg) [inline src/meds_etl/omop.py:197]: Why not:
```
            options = table_details.get("time_field_options", []) + ["_start_datetime", "_start_date", "_datetime", "_date"]
            options = [
                cast_to_datetime(schema, table_name + option, move_to_end_of_day=True)
                for option in options
                if table_name + option in schema.names()
            ]
```

**Review by EthanSteinberg — APPROVED** [review]
Thanks! This looks good with just one minor suggestion for improvement, but you can ignore it if you want

File: src/meds_etl/omop.py (hunk 1 above)
Comment (ChaoPang) [reply src/meds_etl/omop.py:197]: @EthanSteinberg I like your suggestion cos it's more concise. Actually, the fields defined `time_field_options` only contain the table name e.g. `visit_end_datetime` and `visit_end_date`, I can change it to `_end_datetime` and `_end_date` so your suggestion would work, what do you think?

Comment (EthanSteinberg) [reply src/meds_etl/omop.py:197]: Hmm. Yeah that would work. I would also be down with changing the code above to:

ptions = table_details.get("time_field_options", []) + [table_name + "_start_datetime", table_name + "_start_date", table_name + "_datetime", table_name + "_date"]

**Review by EthanSteinberg — APPROVED** [review]
All looks good. Thanks especially for the unit test. We really need tests for all the logic in here, but not in this PR

**Comment by EthanSteinberg** [comment]
@ChaoPang  Simply merge this when you are ready

**Pull request merged**